patterns/1-initial/capacity-for-contributions.md

```diff
@@ -0,0 +1,69 @@
+## Title
+
+Managing capacity for reviewing contributions
+
+## Patlet
+
+Reviewing InnerSource contributions takes time and effort. This should be reflected in capacity planning, especially for larger contributions. Expectations and available capacity should be transparent so that contributors understand when their contributions will be reviewed and released.
+
+## Problem
+
+Large InnerSource contributions are causing delays to other work in the team and/or contributions are taking longer to be released than expected. Reviewing contributions may be significant invisible work, not tracked in a team's agile development process.
+
+## Story
+
+The BBC's connected TV application are built by a number of teams, each with different areas of responsibility. They work on each other's areas of the codebase via InnerSource on a regular basis.
+
+The team responsible for the build process for the JavaScript bundles received a major pull request, changing how dependencies were bundled. This PR introduced a new build time dependency, a new structure to the deployed JavaScript bundles, and touched 503 files, with 6,699 lines of code added and 2,828 lines of code deleted. A contribution of this size required significant time to code review, test, and ensure the team understood the new tooling and structure introduced. Normally, InnerSource contributions would be reviewed ad-hoc, but the team estimated that this review process would take days rather than hours. Reviewing this PR would have delayed the team's other work, so the team raised this with the team lead, delivery manager, and product manager, to prioritize this work against other work. Time was set aside to review this contribution at future date.
+
+This process was formalized in the team:
+
+* Larger contributions have a ticket created on the team's backlog and included in prioritization discussion alongside other tickets. The contributor will be informed of the priority call and given an estimate as to when it will be reviewed and released.
+* Smaller contributions can still be reviewed ad-hoc.
+
+## Context
+
+Maintainers of a successful InnerSource project are finding it difficult to review contributions, especially large contributions. They do not want to disrupt their team's other work, but also want to support contributions by reviewing/releasing them in a timely fashion.
+
+## Forces
+
+* Contributors expect timely feedback on their contributions
+* Maintaining team are expected to deliver other work alongside reviewing contributions
+* Missing communication between contributors and maintainers on expectations/lead time for contributions to be reviewed/released
+* Tension in prioritizing InnerSource contributions against other work
+
+## Solutions
```

Comment:
It might be a good idea to recommend that the contributing team informs the receiving team as early as possible when planning large contributions. One form to operationalize this is through Draft PRs, which allows early feedback, specially if the PR text contain a summary of the planned changes.

Using stacked PRs and slicing the reviews into smaller parts could also help.

Reply:
What I have seen many host teams do is to ask for an issue before creating a PR.

Or in other words, starting the conversation about the **why** and **what** (in an issue) before sending the **how** (in a PR).

We could add a note about this here, if it fits.

And I am even starting to think that this point could work as a new tiny pattern as well a la "Remember to talk to each other before sending code" :)

Reply:
@dellagustin-sap with "stacked PRs" you mean breaking up a larger contribution into multiple PRs that should be reviewed in a specific sequence? i.e. first PR1, then PR2, then PR3, ...?

Reply:
@spier , I'm not sure that there is a well understood definition for the term "stacked PRs", but what I meant was something like:
1. Create a DRAFT PR based on a feature branch explaining the change that is going to be sent
2. Instead of committing your changes in one go to the new feature branch, send small incremental PRs to the new feature branch, as soon as you have new code that can be reviewed, this way the team can review small pieces until the feature is ready to be merged as a whole

This only works of course if the change can be split into small increments. That's not always the case.

Reply:
> It might be a good idea to recommend that the contributing team informs the receiving team as early as possible when planning large contributions.

Yes I will add this. There are a few methods I think worth mentioning: drafts PRs (per @dellagustin-sap), issues (per @spier), RFCs, or even an informal conversation e.g. in Slack

> small incremental PRs

Definitely agree this is preferable but goes a little beyond the scope of this pattern. Small increments is IMO best practice for Agile, InnerSource, CD... Perhaps I'll mention this in 'forces' - something like 'contributors already strive to make small PRs, but find instances where larger PRs are unavoidable.'

Reply:
https://github.com/InnerSourceCommons/InnerSourcePatterns/pull/692/commits/f4bdb57642afca998839c35016ed78f2f60e3fe3

https://github.com/InnerSourceCommons/InnerSourcePatterns/pull/692/commits/d9d5e284a5820b3f6fb2d82767f0326d417117b7

Reply:
Amazing changes @tsadler1988!

@dellagustin-sap as you started this particular thread, would you mind giving this a review? And  while you are at it, possibly even leaving an official review on the PR, to confirm that I am not the only one approving it.

Cheers :)

Reply:
IIRC we even mention the "please talk to the receiving team before submitting a PR" part in the learning path :)

It's one of the things that I keep telling people even in open source: Don't send a PR as your first line of communication. Nothing is more frustrating than going though learning the project, making the modification work and then being told that the receiving team does not accept it - no matter how good the reason may be.

Reply:
I believe the comments in this thread have been worked into the content. Resolving the thread.